### PR TITLE
Feedback

### PR DESCRIPTION
**Pull request by iakov**
:wave:! GitHub Classroom created this pull request as a place for your teacher to leave feedback on your work. It will update automatically. **Don’t close or merge this pull request**, unless you’re instructed to do so by your teacher.

In this pull request, your teacher can leave comments and feedback on your code. Click the **Subscribe** button to be notified if that happens.

Click the **Files changed** or **Commits** tab to see all of the changes pushed to `master` since the assignment started. Your teacher can see this too.

<details>
<summary>
<strong>Notes for teachers</strong>
</summary>

Use this PR to leave feedback. Here are some tips:

- Click the **Files changed** tab to see all of the changes pushed to `master` since the assignment started. To leave comments on specific lines of code, put your cursor over a line of code and click the blue **+** (plus sign). To learn more about comments, read “[Commenting on a pull request](https://docs.github.com/en/github/collaborating-with-issues-and-pull-requests/commenting-on-a-pull-request)”.
- Click the **Commits** tab to see the commits pushed to `master`. Click a commit to see specific changes.
- If you turned on autograding, then click the **Checks** tab to see the results.
- This page is an overview. It shows commits, line comments, and general comments. You can leave a general comment below.

For more information about this pull request, read “[Leaving assignment feedback in GitHub](https://docs.github.com/education/manage-coursework-with-github-classroom/leave-feedback-with-pull-requests)”.
</details>

Subscribed: @IlyaMuravjov

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -0,0 +1,67 @@
...
+
+all: $(TARGET)
+
+$(OBJECT_FILES): $(OBJECT_DIR)/%.o: $(SOURCE_DIR)/%.c $(CONFIG) | $(OBJECT_DIR)
+	$(CC) $(CFLAGS) -c $< -o $@
+
+$(OBJECT_DIR)/main.o: $(SOURCE_DIR)/priority_queue.h $(SOURCE_DIR)/logging.h
+$(OBJECT_DIR)/priority_queue.o: $(SOURCE_DIR)/priority_queue.h $(SOURCE_DIR)/logging.h
+
+$(OBJECT_DIR): ; mkdir -p $@
```
Comment (KutuevVladimir) [inline Makefile:33]: В чём причина использования различных стилей написания recipe?

File: Makefile [hunk 2]
```diff
@@ -0,0 +1,67 @@
...
+	@echo "Test $* $$(cat $(TEST_RESULT_DIR)/$*.log)"
+
+$(TEST_LOG_FILES): $(TEST_RESULT_DIR)/%.log: $(TEST_DIR)/%.out $(TEST_RESULT_DIR)/%.actual $(CONFIG) | $(TEST_RESULT_DIR)
+	@if cmp "$(TEST_DIR)/$*.out" "$(TEST_RESULT_DIR)/$*.actual"; then \
+    	echo -n $(PASSED_MSG) > "$@"; \
+     else \
+       	echo -n $(FAILED_MSG) > "$@"; \
+     fi; \
+
+$(TEST_ACTUAL_FILES): $(TEST_RESULT_DIR)/%.actual: $(TEST_DIR)/%.in $(TARGET) $(CONFIG) | $(TEST_RESULT_DIR)
```
Comment (KutuevVladimir) [inline Makefile:61]: Чем обусловлено использование Static Pattern Rules?

**Review by KutuevVladimir — COMMENTED** [review]
Очень круто сделан вывод информации о тестах. С другой стороны такой Makefile лично мне кажется слишком тяжёлым для такой маленькой задачи.
Но в целом всё ок. Оценка 10 баллов.

File: Makefile (hunk 1 above)
Comment (IlyaMuravjov) [reply Makefile:33]: Стиль `targets : prerequisites ; recipe` использовался для правил без `prerequisites` с очень коротким `recipe` для улучшения визуального восприятия кода. 

В остальных случаях использовался стиль:
```
targets : prerequisites
        recipe
        …
```

**Pull request merged**